### PR TITLE
Add RealTimeMicroseconds to HTTP Header X-ClickHouse-Summary

### PR DESCRIPTION
**Pull request by alexon1234**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Adding `RealTimeMicroseconds` metric in the HTTP Header `X-ClickHouse-Summary`. This way we can know the CPU time of a request without having to go to check it in `system.query_log`

```
» curl "http://localhost:8123?query=SELECT%20uniq%28number%29%20FROM%20numbers%2820%29&wait_end_of_query=1&log_queries=1"  -vv                                   [4:49:20]
*   Trying 127.0.0.1:8123...
* Connected to localhost (127.0.0.1) port 8123 (#0)
> GET /?query=SELECT%20uniq%28number%29%20FROM%20numbers%2820%29&wait_end_of_query=1&log_queries=1 HTTP/1.1
> Host: localhost:8123
> User-Agent: curl/7.81.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Date: Mon, 26 Aug 2024 15:50:54 GMT
< Connection: Keep-Alive
< Content-Type: text/tab-separated-values; charset=UTF-8
< X-ClickHouse-Server-Display-Name: alejandro-XPS-15-9500
< Transfer-Encoding: chunked
< X-ClickHouse-Query-Id: 9c0eb3ab-9a7a-4d3d-abcb-f1eca5c67986
< X-ClickHouse-Format: TabSeparated
< X-ClickHouse-Timezone: Atlantic/Canary
< Keep-Alive: timeout=30
< X-ClickHouse-Summary: {"read_rows":"20","read_bytes":"160","written_rows":"0","written_bytes":"0","total_rows_to_read":"20","result_rows":"1","result_bytes":"136","elapsed_ns":"3968563","real_time_microseconds":"18002"}
< 
20
* Connection #0 to host localhost left intact
```

```
 :) SELECT * FROM system.query_log WHERE query_id = '9c0eb3ab-9a7a-4d3d-abcb-f1eca5c67986' AND type >1 FORMAT Vertical

SELECT *
FROM system.query_log
WHERE (query_id = '9c0eb3ab-9a7a-4d3d-abcb-f1eca5c67986') AND (type > 1)
FORMAT Vertical

Row 1:
──────
...
query:                                 SELECT uniq(number) FROM numbers(20)
ProfileEvents:                         {'Query':1,'SelectQuery':1,'InitialQuery':1,'QueriesWithSubqueries':1,'SelectQueriesWithSubqueries':1,'WriteBufferFromFileDescriptorWrite':1,'WriteBufferFromFileDescriptorWriteBytes':3,'IOBufferAllocs':4,'IOBufferAllocBytes':4194556,'ArenaAllocChunks':1,'ArenaAllocBytes':4096,'TableFunctionExecute':1,'DiskWriteElapsedMicroseconds':19,'SelectedRows':20,'SelectedBytes':160,'ContextLock':20,'RealTimeMicroseconds':18002,'UserTimeMicroseconds':1620,'SystemTimeMicroseconds':1518,'SoftPageFaults':1330,'OSCPUWaitMicroseconds':7,'OSCPUVirtualTimeMicroseconds':3120,'OSWriteBytes':4096,'OSReadChars':7733,'OSWriteChars':2162,'LogTrace':9,'LogDebug':1}
``` 

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 6b08d2b6de2f22b5972a0b27e6f77ab98344a1da with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/69032/6b08d2b6de2f22b5972a0b27e6f77ab98344a1da/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/69032/6b08d2b6de2f22b5972a0b27e6f77ab98344a1da/style_check.html">❌ failure</a></td></tr>
<tbody>
</table>

File: src/Interpreters/executeQuery.cpp [hunk 1]
```diff
@@ -419,6 +419,9 @@ void logQueryFinish(
         {
             Progress p;
             p.incrementPiecewiseAtomically(Progress{ResultProgress{elem.result_rows, elem.result_bytes}});
+
+            UInt64 real_time_microseconds = (*info.profile_counters)[ProfileEvents::RealTimeMicroseconds];
```
Comment (yakov-olkhovskiy) [inline src/Interpreters/executeQuery.cpp:423]: you need to define `ProfileEvents::RealTimeMicroseconds` as extern:
```
namespace ProfileEvents
{
    extern const Event RealTimeMicroseconds;
}
```

Comment (alexon1234) [reply src/Interpreters/executeQuery.cpp:424]: Thanks

**Comment by yakov-olkhovskiy** [comment]
@alexon1234 you have segfault:
```
2024.08.28 22:10:45.356313 [ 166045 ] {} <Trace> BaseDaemon: Received signal 11
2024.08.28 22:10:45.356500 [ 367521 ] {} <Fatal> BaseDaemon: ########## Short fault info ############
2024.08.28 22:10:45.356700 [ 367521 ] {} <Fatal> BaseDaemon: (version 24.9.1.1, build id: D8BDC8194AEA8208FBF4EEA1B512F4DFF0CB961A, git hash: 8f650190fea42d5300143452042bf53770d881fd) (from thread 172822) Received signal 11
2024.08.28 22:10:45.356716 [ 367521 ] {} <Fatal> BaseDaemon: Signal description: Segmentation fault
2024.08.28 22:10:45.356729 [ 367521 ] {} <Fatal> BaseDaemon: Address: NULL pointer. Access: read. Address not mapped to object.
2024.08.28 22:10:45.356743 [ 367521 ] {} <Fatal> BaseDaemon: Stack trace: 0x000000000802b588 0x00000000082af8f0 0x000000000e5cc3c2 0x000000000c70d6db 0x000000000c71c4e0 0x000000000baf696e 0x000000000d98ce55 0x000000000d9a2199 0x000000000e2a1667 0x000000000e2a1b3a 0x000000000e252e12 0x000000000e250903
2024.08.28 22:10:45.356756 [ 367521 ] {} <Fatal> BaseDaemon: ########################################
2024.08.28 22:10:45.356821 [ 367521 ] {} <Fatal> BaseDaemon: (version 24.9.1.1, build id: D8BDC8194AEA8208FBF4EEA1B512F4DFF0CB961A, git hash: 8f650190fea42d5300143452042bf53770d881fd) (from thread 172822) (query_id: 96e0e553-85e8-4886-8a79-a035eadafa4e) (query: SET DEFAULT ROLE sqllt_role TO sqllt_user;) Received signal Segmentation fault (11)
2024.08.28 22:10:45.356896 [ 367521 ] {} <Fatal> BaseDaemon: Address: NULL pointer. Access: read. Address not mapped to object.
2024.08.28 22:10:45.356936 [ 367521 ] {} <Fatal> BaseDaemon: Stack trace: 0x000000000802b588 0x00000000082af8f0 0x000000000e5cc3c2 0x000000000c70d6db 0x000000000c71c4e0 0x000000000baf696e 0x000000000d98ce55 0x000000000d9a2199 0x000000000e2a1667 0x000000000e2a1b3a 0x000000000e252e12 0x000000000e250903
2024.08.28 22:10:45.367093 [ 367521 ] {} <Fatal> BaseDaemon: 0.0. inlined from /fasttest-workspace/build/./src/Common/StackTrace.cpp:372: StackTrace::tryCapture()
2024.08.28 22:10:45.367135 [ 367521 ] {} <Fatal> BaseDaemon: 0. /fasttest-workspace/build/./src/Common/StackTrace.cpp:341: StackTrace::StackTrace(ucontext const&) @ 0x000000000802b588
2024.08.28 22:10:45.382753 [ 367521 ] {} <Fatal> BaseDaemon: 1. /fasttest-workspace/build/./src/Common/SignalHandlers.cpp:84: signalHandler(int, siginfo_t*, void*) @ 0x00000000082af8f0
2024.08.28 22:10:45.394881 [ 367521 ] {} <Fatal> BaseDaemon: 2. __restore_rt @ 0x000000000e5cc3c2
2024.08.28 22:10:45.453077 [ 367521 ] {} <Fatal> BaseDaemon: 3.0. inlined from ./contrib/llvm-project/libcxx/include/__memory/unique_ptr.h:492: std::unique_ptr<unsigned long [], std::default_delete<unsigned long []>>::operator[][abi:v15007](unsigned long) const
2024.08.28 22:10:45.453111 [ 367521 ] {} <Fatal> BaseDaemon: 3.1. inlined from ./src/Common/ProfileEvents.h:94: ProfileEvents::Counters::Snapshot::operator[](StrongTypedef<unsigned long, ProfileEvents::EventTag>) const
2024.08.28 22:10:45.453138 [ 367521 ] {} <Fatal> BaseDaemon: 3. /fasttest-workspace/build/./src/Interpreters/executeQuery.cpp:424: DB::logQueryFinish(DB::QueryLogElement&, std::shared_ptr<DB::Context> const&, std::shared_ptr<DB::IAST> const&, DB::QueryPipeline const&, bool, std::shared_ptr<DB::OpenTelemetry::SpanHolder>, DB::QueryCache::Usage, bool) @ 0x000000000c70d6db
2024.08.28 22:10:45.511694 [ 367521 ] {} <Fatal> BaseDaemon: 4.0. inlined from /fasttest-workspace/build/./src/Interpreters/executeQuery.cpp:1348: operator()
2024.08.28 22:10:45.511719 [ 367521 ] {} <Fatal> BaseDaemon: 4.1. inlined from ./contrib/llvm-project/libcxx/include/__functional/invoke.h:394: ?
2024.08.28 22:10:45.511763 [ 367521 ] {} <Fatal> BaseDaemon: 4.2. inlined from ./contrib/llvm-project/libcxx/include/__functional/invoke.h:479: ?
2024.08.28 22:10:45.511802 [ 367521 ] {} <Fatal> BaseDaemon: 4.3. inlined from ./contrib/llvm-project/libcxx/include/__functional/function.h:235: ?
2024.08.28 22:10:45.511832 [ 367521 ] {} <Fatal> BaseDaemon: 4. ./contrib/llvm-project/libcxx/include/__functional/function.h:716: ? @ 0x000000000c71c4e0
2024.08.28 22:10:45.514873 [ 367521 ] {} <Fatal> BaseDaemon: 5.0. inlined from ./contrib/llvm-project/libcxx/include/__functional/function.h:848: ?
2024.08.28 22:10:45.514892 [ 367521 ] {} <Fatal> BaseDaemon: 5.1. inlined from ./contrib/llvm-project/libcxx/include/__functional/function.h:1197: ?
2024.08.28 22:10:45.514936 [ 367521 ] {} <Fatal> BaseDaemon: 5. /fasttest-workspace/build/./src/QueryPipeline/BlockIO.cpp:57: DB::BlockIO::onFinish() @ 0x000000000baf696e
2024.08.28 22:10:45.549881 [ 367521 ] {} <Fatal> BaseDaemon: 6.0. inlined from /fasttest-workspace/build/./src/Server/TCPHandler.cpp:574: operator()
2024.08.28 22:10:45.549912 [ 367521 ] {} <Fatal> BaseDaemon: 6. /fasttest-workspace/build/./src/Server/TCPHandler.cpp:0: DB::TCPHandler::runImpl() @ 0x000000000d98ce55
2024.08.28 22:10:45.608640 [ 367521 ] {} <Fatal> BaseDaemon: 7. /fasttest-workspace/build/./src/Server/TCPHandler.cpp:2484: DB::TCPHandler::run() @ 0x000000000d9a2199
2024.08.28 22:10:45.609791 [ 367521 ] {} <Fatal> BaseDaemon: 8. /fasttest-workspace/build/./base/poco/Net/src/TCPServerConnection.cpp:43: Poco::Net::TCPServerConnection::start() @ 0x000000000e2a1667
2024.08.28 22:10:45.611845 [ 367521 ] {} <Fatal> BaseDaemon: 9. /fasttest-workspace/build/./base/poco/Net/src/TCPServerDispatcher.cpp:115: Poco::Net::TCPServerDispatcher::run() @ 0x000000000e2a1b3a
2024.08.28 22:10:45.614385 [ 367521 ] {} <Fatal> BaseDaemon: 10. /fasttest-workspace/build/./base/poco/Foundation/src/ThreadPool.cpp:205: Poco::PooledThread::run() @ 0x000000000e252e12
2024.08.28 22:10:45.616709 [ 367521 ] {} <Fatal> BaseDaemon: 11. ./base/poco/Foundation/src/Thread_POSIX.cpp:335: Poco::ThreadImpl::runnableEntry(void*) @ 0x000000000e250903
2024.08.28 22:10:45.616734 [ 367521 ] {} <Fatal> BaseDaemon: Integrity check of the executable skipped because the reference checksum could not be read.
2024.08.28 22:10:45.875916 [ 367521 ] {} <Fatal> BaseDaemon: This ClickHouse version is not official and should be upgraded to the official build.
2024.08.28 22:10:45.876011 [ 166721 ] {} <Trace> SystemLog (system.crash_log): Flushing system log, 1 entries to flush up to offset 1
2024.08.28 22:10:45.876105 [ 367521 ] {} <Fatal> BaseDaemon: Changed settings: connect_timeout_with_failover_ms = 2000, connect_timeout_with_failover_secure_ms = 3000, idle_connection_timeout = 36000, s3_max_get_rps = 1000000, s3_max_get_burst = 2000000, s3_max_put_rps = 1000000, s3_max_put_burst = 2000000, s3_check_objects_after_upload = true, max_remote_read_network_bandwidth = 1000000000000, max_remote_write_network_bandwidth = 1000000000000, max_local_read_bandwidth = 1000000000000, max_local_write_bandwidth = 1000000000000, stream_like_engine_allow_direct_select = true, replication_wait_for_inactive_replica_timeout = 30, allow_nonconst_timezone_arguments = true, log_queries = true, insert_quorum_timeout = 60000, fsync_metadata = false, http_send_timeout = 60., http_receive_timeout = 60., use_index_for_in_with_subqueries_max_values = 1000000000, opentelemetry_start_trace_probability = 0.10000000149011612, max_rows_to_read = 20000000, max_bytes_to_read = 1000000000000, max_bytes_to_read_leaf = 1000000000000, max_rows_to_group_by = 10000000000, max_bytes_before_external_group_by = 10000000000, max_rows_to_sort = 10000000000, max_bytes_to_sort = 10000000000, max_bytes_before_external_sort = 10000000000, max_result_rows = 1000000000, max_result_bytes = 1000000000, max_execution_time = 600., max_execution_time_leaf = 600., max_execution_speed = 100000000000, max_execution_speed_bytes = 10000000000000, timeout_before_checking_execution_speed = 300., max_estimated_execution_time = 600., max_columns_to_read = 20000, max_temporary_columns = 20000, max_temporary_non_const_columns = 20000, max_rows_in_set = 10000000000, max_bytes_in_set = 10000000000, max_rows_in_join = 10000000000, max_bytes_in_join = 10000000000, max_rows_to_transfer = 1000000000, max_bytes_to_transfer = 1000000000, max_rows_in_distinct = 10000000000, max_bytes_in_distinct = 10000000000, max_memory_usage = 5000000000, max_memory_usage_for_user = 32000000000, max_untracked_memory = 1048576, memory_profiler_step = 1048576, max_network_bandwidth = 100000000000, max_network_bytes = 1000000000000, max_network_bandwidth_for_user = 100000000000, max_network_bandwidth_for_all_users = 100000000000, max_temporary_data_on_disk_size_for_user = 100000000000, max_temporary_data_on_disk_size_for_query = 100000000000, max_backup_bandwidth = 100000000000, log_profile_events = false, log_comment = 'system.query_log logging test', send_logs_level = 'warning', max_hyperscan_regexp_length = 1000000, max_hyperscan_regexp_total_length = 10000000, allow_introspection_functions = true, database_atomic_wait_for_drop_and_detach_synchronously = true, query_cache_max_size_in_bytes = 10000000, query_cache_max_entries = 100000, distributed_ddl_entry_format_version = 6, external_storage_max_read_rows = 10000000000, external_storage_max_read_bytes = 10000000000, async_insert_busy_timeout_max_ms = 5000, load_marks_asynchronously = true, allow_prefetched_read_pool_for_remote_filesystem = false, allow_prefetched_read_pool_for_local_filesystem = false, filesystem_prefetch_max_memory_usage = 1073741824, max_streams_for_merge_tree_reading = 1000, insert_keeper_max_retries = 100, insert_keeper_retry_initial_backoff_ms = 1, insert_keeper_retry_max_backoff_ms = 10, insert_keeper_fault_injection_probability = 0.009999999776482582
```
because `info.profile_counters` is not initialized, because it depends on `log_profile_events` setting used in executeQuery.cpp:402
```
        QueryStatusInfo info = process_list_elem->getInfo(true, context->getSettingsRef().log_profile_events);
```

File: src/Interpreters/executeQuery.cpp [hunk 2]
```diff
@@ -419,6 +420,13 @@ void logQueryFinish(
         {
             Progress p;
             p.incrementPiecewiseAtomically(Progress{ResultProgress{elem.result_rows, elem.result_bytes}});
+
+
+            if (info.profile_counters)
+            {
+                UInt64 real_time_microseconds = (*info.profile_counters)[ProfileEvents::RealTimeMicroseconds];
+                p.incrementRealTimeMicroseconds(real_time_microseconds);
+            }
```
Comment (yakov-olkhovskiy) [inline src/Interpreters/executeQuery.cpp:429]: This is not very good solution to have this dependent from irrelevant `log_profile_events` setting. Let's replace line 402 with:
```
        std::shared_ptr<ProfileEvents::Counters::Snapshot> profile_counters;

        QueryStatusInfo info = process_list_elem->getInfo(true, true);
        if (context->getSettingsRef().log_profile_events)
            profile_counters = info.profile_counters;
        else
            profile_counters.swap(info.profile_counters);
```
and this code with:
```
            p.incrementRealTimeMicroseconds(profile_counters[ProfileEvents::RealTimeMicroseconds]);
```

Comment (alexon1234) [reply src/Interpreters/executeQuery.cpp]: Thanks for the suggestion

**Comment by alexon1234** [comment]
@yakov-olkhovskiy I've checked the CI, especially the tests that are reported to fail and seem not related to my changes. 
Checking the URL https://s3.amazonaws.com/clickhouse-test-reports/69032/36725fb5e149ed532aca6c290585dec444ed03b4/stateless_tests__asan__[1_2].html, I can see the test `Server died` has failed

Also, checking https://s3.amazonaws.com/clickhouse-test-reports/69032/36725fb5e149ed532aca6c290585dec444ed03b4/stateless_tests__asan__[2_2].html, reports that the test `02944_variant_as_common_type` is broken. I'm taking a look at that

**Comment by alexon1234** [comment]
> @yakov-olkhovskiy I've checked the CI, especially the tests that are reported to fail and seem not related to my changes. Checking the URL [https://s3.amazonaws.com/clickhouse-test-reports/69032/36725fb5e149ed532aca6c290585dec444ed03b4/stateless_tests__asan__[1_2].html](https://s3.amazonaws.com/clickhouse-test-reports/69032/36725fb5e149ed532aca6c290585dec444ed03b4/stateless_tests__asan__%5B1_2%5D.html), I can see the test `Server died` has failed. Seems this test fails from time to time
> 
> Also, checking [https://s3.amazonaws.com/clickhouse-test-reports/69032/36725fb5e149ed532aca6c290585dec444ed03b4/stateless_tests__asan__[2_2].html](https://s3.amazonaws.com/clickhouse-test-reports/69032/36725fb5e149ed532aca6c290585dec444ed03b4/stateless_tests__asan__%5B2_2%5D.html), reports that the test `02944_variant_as_common_type` is broken. I'm taking a look at that
> 
> Running the test against the head version of docker, I can see the test reports the same error
> 
> ```diff
>  » PATH=./build/programs/clickhouse-client:$PATH tests/clickhouse-test 02944_variant_as_common_type                                                                  
> Using queries from /ClickHouse/tests/queries' directory
> Connecting to ClickHouse server... OK
> Connected to server 24.9.1.1535 @ b3dae16fd4d3f5364dc5ee63a20beb435a27de90 HEAD
> Running 2 stateless tests (MainProcess).
> 02944_variant_as_common_type:                                           [ FAIL ]
> Reason: result differs with reference:  
> --- tests/queries/0_stateless/02944_variant_as_common_type.reference	2024-02-03 11:05:17.869225230 +0000
> +++ tests/queries/0_stateless/02944_variant_as_common_type.stdout	2024-08-29 18:21:39.321739565 +0100
> @@ -1,51 +1,51 @@
> -Array(UInt8)	[1,2,3]
> -Array(UInt8)	[1,2,3]
> -String	str_1
> -Nullable(String)	str_1
> -String	str_1
> -Nullable(String)	str_1
> -Variant(Array(UInt8), String)	str_1
> -Variant(Array(UInt8), String)	str_1
> -Array(UInt8)	[1,2,3]
> -Array(UInt8)	[1,2,3]
> -String	str_1
> -Nullable(String)	str_1
> -String	str_1
> -Nullable(String)	str_1
> -Variant(Array(UInt8), String)	str_1
> -Variant(Array(UInt8), String)	str_1
> -Array(UInt8)	[1,2,3]
> -Array(UInt8)	[1,2,3]
> -String	str_1
> -Nullable(String)	str_1
> -String	str_1
> -Nullable(String)	str_1
> -Variant(Array(UInt8), String)	str_1
> -Variant(Array(UInt8), String)	str_1
> -String	str_0
> -String	str_1
> -String	str_2
> -String	str_3
> -Nullable(String)	str_0
> -Nullable(String)	str_1
> -Nullable(String)	str_2
> -Nullable(String)	str_3
> -Array(UInt64)	[0]
> -Array(UInt64)	[0,1]
> -Array(UInt64)	[0,1,2]
> -Array(UInt64)	[0,1,2,3]
> -Array(UInt64)	[0]
> -Array(UInt64)	[0,1]
> -Array(UInt64)	[0,1,2]
> -Array(UInt64)	[0,1,2,3]
> -String	str_0
> -String	str_1
> -String	str_2
> -String	str_3
> -Nullable(String)	str_0
> -Nullable(String)	str_1
> -Nullable(String)	str_2
> -Nullable(String)	str_3
> +Variant(Array(UInt8), String)	[1,2,3]
> +Variant(Array(UInt8), String)	[1,2,3]
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt8), String)	[1,2,3]
> +Variant(Array(UInt8), String)	[1,2,3]
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt8), String)	[1,2,3]
> +Variant(Array(UInt8), String)	[1,2,3]
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt8), String)	str_1
> +Variant(Array(UInt64), String)	str_0
> +Variant(Array(UInt64), String)	str_1
> +Variant(Array(UInt64), String)	str_2
> +Variant(Array(UInt64), String)	str_3
> +Variant(Array(UInt64), String)	str_0
> +Variant(Array(UInt64), String)	str_1
> +Variant(Array(UInt64), String)	str_2
> +Variant(Array(UInt64), String)	str_3
> +Variant(Array(UInt64), String)	[0]
> +Variant(Array(UInt64), String)	[0,1]
> +Variant(Array(UInt64), String)	[0,1,2]
> +Variant(Array(UInt64), String)	[0,1,2,3]
> +Variant(Array(UInt64), String)	[0]
> +Variant(Array(UInt64), String)	[0,1]
> +Variant(Array(UInt64), String)	[0,1,2]
> +Variant(Array(UInt64), String)	[0,1,2,3]
> +Variant(Array(UInt64), String)	str_0
> +Variant(Array(UInt64), String)	str_1
> +Variant(Array(UInt64), String)	str_2
> +Variant(Array(UInt64), String)	str_3
> +Variant(Array(UInt64), String)	str_0
> +Variant(Array(UInt64), String)	str_1
> +Variant(Array(UInt64), String)	str_2
> +Variant(Array(UInt64), String)	str_3
>  Variant(Array(UInt64), String)	str_0
>  Variant(Array(UInt64), String)	str_1
>  Variant(Array(UInt64), String)	str_2
> ```

Taking a look this tests should be run with the old analyzer

**Comment by alexon1234** [comment]
> > @yakov-olkhovskiy I've checked the CI, especially the tests that are reported to fail and seem not related to my changes. Checking the URL [https://s3.amazonaws.com/clickhouse-test-reports/69032/36725fb5e149ed532aca6c290585dec444ed03b4/stateless_tests__asan__[1_2].html](https://s3.amazonaws.com/clickhouse-test-reports/69032/36725fb5e149ed532aca6c290585dec444ed03b4/stateless_tests__asan__%5B1_2%5D.html), I can see the test `Server died` has failed. Seems this test fails from time to time
> > Also, checking [https://s3.amazonaws.com/clickhouse-test-reports/69032/36725fb5e149ed532aca6c290585dec444ed03b4/stateless_tests__asan__[2_2].html](https://s3.amazonaws.com/clickhouse-test-reports/69032/36725fb5e149ed532aca6c290585dec444ed03b4/stateless_tests__asan__%5B2_2%5D.html), reports that the test `02944_variant_as_common_type` is broken. I'm taking a look at that
> > Running the test against the head version of docker, I can see the test reports the same error
> > ```diff
> >  » PATH=./build/programs/clickhouse-client:$PATH tests/clickhouse-test 02944_variant_as_common_type                                                                  
> > Using queries from /ClickHouse/tests/queries' directory
> > Connecting to ClickHouse server... OK
> > Connected to server 24.9.1.1535 @ b3dae16fd4d3f5364dc5ee63a20beb435a27de90 HEAD
> > Running 2 stateless tests (MainProcess).
> > 02944_variant_as_common_type:                                           [ FAIL ]
> > Reason: result differs with reference:  
> > --- tests/queries/0_stateless/02944_variant_as_common_type.reference	2024-02-03 11:05:17.869225230 +0000
> > +++ tests/queries/0_stateless/02944_variant_as_common_type.stdout	2024-08-29 18:21:39.321739565 +0100
> > @@ -1,51 +1,51 @@
> > -Array(UInt8)	[1,2,3]
> > -Array(UInt8)	[1,2,3]
> > -String	str_1
> > -Nullable(String)	str_1
> > -String	str_1
> > -Nullable(String)	str_1
> > -Variant(Array(UInt8), String)	str_1
> > -Variant(Array(UInt8), String)	str_1
> > -Array(UInt8)	[1,2,3]
> > -Array(UInt8)	[1,2,3]
> > -String	str_1
> > -Nullable(String)	str_1
> > -String	str_1
> > -Nullable(String)	str_1
> > -Variant(Array(UInt8), String)	str_1
> > -Variant(Array(UInt8), String)	str_1
> > -Array(UInt8)	[1,2,3]
> > -Array(UInt8)	[1,2,3]
> > -String	str_1
> > -Nullable(String)	str_1
> > -String	str_1
> > -Nullable(String)	str_1
> > -Variant(Array(UInt8), String)	str_1
> > -Variant(Array(UInt8), String)	str_1
> > -String	str_0
> > -String	str_1
> > -String	str_2
> > -String	str_3
> > -Nullable(String)	str_0
> > -Nullable(String)	str_1
> > -Nullable(String)	str_2
> > -Nullable(String)	str_3
> > -Array(UInt64)	[0]
> > -Array(UInt64)	[0,1]
> > -Array(UInt64)	[0,1,2]
> > -Array(UInt64)	[0,1,2,3]
> > -Array(UInt64)	[0]
> > -Array(UInt64)	[0,1]
> > -Array(UInt64)	[0,1,2]
> > -Array(UInt64)	[0,1,2,3]
> > -String	str_0
> > -String	str_1
> > -String	str_2
> > -String	str_3
> > -Nullable(String)	str_0
> > -Nullable(String)	str_1
> > -Nullable(String)	str_2
> > -Nullable(String)	str_3
> > +Variant(Array(UInt8), String)	[1,2,3]
> > +Variant(Array(UInt8), String)	[1,2,3]
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt8), String)	[1,2,3]
> > +Variant(Array(UInt8), String)	[1,2,3]
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt8), String)	[1,2,3]
> > +Variant(Array(UInt8), String)	[1,2,3]
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt8), String)	str_1
> > +Variant(Array(UInt64), String)	str_0
> > +Variant(Array(UInt64), String)	str_1
> > +Variant(Array(UInt64), String)	str_2
> > +Variant(Array(UInt64), String)	str_3
> > +Variant(Array(UInt64), String)	str_0
> > +Variant(Array(UInt64), String)	str_1
> > +Variant(Array(UInt64), String)	str_2
> > +Variant(Array(UInt64), String)	str_3
> > +Variant(Array(UInt64), String)	[0]
> > +Variant(Array(UInt64), String)	[0,1]
> > +Variant(Array(UInt64), String)	[0,1,2]
> > +Variant(Array(UInt64), String)	[0,1,2,3]
> > +Variant(Array(UInt64), String)	[0]
> > +Variant(Array(UInt64), String)	[0,1]
> > +Variant(Array(UInt64), String)	[0,1,2]
> > +Variant(Array(UInt64), String)	[0,1,2,3]
> > +Variant(Array(UInt64), String)	str_0
> > +Variant(Array(UInt64), String)	str_1
> > +Variant(Array(UInt64), String)	str_2
> > +Variant(Array(UInt64), String)	str_3
> > +Variant(Array(UInt64), String)	str_0
> > +Variant(Array(UInt64), String)	str_1
> > +Variant(Array(UInt64), String)	str_2
> > +Variant(Array(UInt64), String)	str_3
> >  Variant(Array(UInt64), String)	str_0
> >  Variant(Array(UInt64), String)	str_1
> >  Variant(Array(UInt64), String)	str_2
> > ```
> 
> Taking a look these tests should be run with the old analyzer

Fixed test in commit https://github.com/ClickHouse/ClickHouse/pull/69032/commits/aeaaef4347d614bded7dcbe2c230924847ba579a and also I've created the following PR https://github.com/ClickHouse/ClickHouse/pull/69088 to merge the fix before this PR gets merged

File: tests/queries/0_stateless/02944_variant_as_common_type.sql [hunk 3]
```diff
@@ -1,5 +1,6 @@
 set allow_experimental_variant_type=1;
 set use_variant_as_common_type=1;
+set allow_experimental_analyzer=0;
```
Comment (yakov-olkhovskiy) [inline tests/queries/0_stateless/02944_variant_as_common_type.sql]: tests mentioned in `tests/analyzer_tech_debt.txt` can fail, don't fix them

**Comment by yakov-olkhovskiy** [comment]
02735_parquet_encoder with tsan seems to be too long and thus flaky:
https://github.com/ClickHouse/ClickHouse/pull/68984

**Comment by alexey-milovidov** [comment]
This change is highly questionable for the following reasons:

1. The metric `RealTimeMicroseconds` is almost pointless and requires an expert-level understanding to know its meaning (the sum of the wall clock time across all the threads that participated in the query run time). It is not possible to imagine a situation when a user needs it.

2. This metric is not anyhow better or more special than other hundreds of metrics.

3. The code is not particularly generic or elegant and involves manual tracking of the parameters across multiple places of the code base.

4. It introduces a new naming schema for metrics: `real_time_microseconds` instead of `RealTimeMicroseconds,` which will be a burden to support while adding every new metric.

5. A natural extension would be adding more metrics, but it is impossible due to the various limitations on the size of HTTP headers.

6. It looks out of place with the presence of the `elapsed_ns` data.

This pull request will be reverted.

**Comment by yakov-olkhovskiy** [comment]
@alexon1234 maybe we need to find more general approach to retrieve arbitrary metrics within HTTP header - the idea looks valuable to me...

**Comment by alexey-milovidov** [comment]
@yakov-olkhovskiy Introduce the notion of "framing formats".

A framing format wraps chunks of resultset, progress events, and metrics.
As a bonus, it will allow to stream results from multiple queries in a single request, even in parallel.

It is mostly orthogonal to result formats, because it can wrap different result formats.

Examples of framing formats will be:
- none: output results as is, do not wrap;
- application/event-stream: use server-sent events in the HTTP protocol.

**Pull request merged**